### PR TITLE
refactor(connlib): log more details about failed queries

### PR DESCRIPTION
**Pull request by conectado**

### CODE REVIEW

**Comment by ReactorScram** [comment]
Do we have a policy for what info is okay / not okay to log? I was wondering about logging resource URLs / names when doing clipboard copying yesterday and ended up just saying "copied a resource to clipboard"

**Review by jamilbk — APPROVED** [review]

File: rust/connlib/clients/shared/src/control.rs [hunk 1]
```diff
@@ -353,9 +353,9 @@ impl<CB: Callbacks + 'static> ControlPlane<CB> {
                 };
                 let tunnel = self.tunnel.clone();
                 tokio::spawn(async move {
-                    let response = resolver.lookup(query.name, query.record_type).await;
+                    let response = resolver.lookup(&query.name, query.record_type).await;
                     if let Err(err) = tunnel.write_dns_lookup_response(response, query.query) {
-                        tracing::error!(err = ?err, "DNS lookup failed: {err:#}");
+                        tracing::error!(err = ?err, name = query.name, record_type = ?query.record_type, "DNS lookup failed: {err:#}");
```
Comment (jamilbk) [inline rust/connlib/clients/shared/src/control.rs]:
```suggestion
                        tracing::debug!(err = ?err, name = query.name, record_type = ?query.record_type, "DNS lookup failed: {err:#}");
```

**Review by jamilbk — CHANGES_REQUESTED** [review]
Let's not log the DNS names that fail to resolve in they contain anything sensitive. Suggest logging in debug

**Comment by jamilbk** [comment]
> Do we have a policy for what info is okay / not okay to log? I was wondering about logging resource URLs / names when doing clipboard copying yesterday and ended up just saying "copied a resource to clipboard"

* Avoid logging anything sensitive in `info!` or higher levels since that is the default for release clients
* Never log secrets (this should be handled by `Secret` already)

**Comment by conectado** [comment]
> Let's not log the DNS names that fail to resolve in they contain anything sensitive. Suggest logging in debug

I don't think a DNS should ever contain sensitive information also we shouldn't depend on log-level for security

**Comment by ReactorScram** [comment]
Yeah I suppose if the DNS resources are cleartext in the portal DB then they're not secret. It would only be if there's a glitchy DNS client that puts the whole URL into the request.

**Review by jamilbk — APPROVED** [review]

**Pull request merged**